batch/batch/driver/instance_collection/pool.py

```diff
@@ -241,29 +246,31 @@ async def create_instance(
         )
 
     def compute_n_instances_needed(
-        self, ready_cores_mcpu: int, n_instances_by_state: Dict[str, int], region: Optional[str] = None
+        self,
+        ready_cores_mcpu: int,
+        n_instances_by_state: Dict[str, int],
+        live_free_cores_mcpu_by_region: Dict[str, int],
+        regions: List[str],
     ):
         n_live_instances = n_instances_by_state['pending'] + n_instances_by_state['active']
+        n_instances = sum(count for count in n_instances_by_state.values())
 
-        if region is None:
-            live_free_cores_mcpu = self.live_free_cores_mcpu
-        else:
-            live_free_cores_mcpu = self.live_free_cores_mcpu_by_region[region]
+        live_free_cores_mcpu = sum(live_free_cores_mcpu_by_region[region] for region in regions)
 
         instances_needed = (ready_cores_mcpu - live_free_cores_mcpu + (self.worker_cores * 1000) - 1) // (
             self.worker_cores * 1000
         )
         instances_needed = min(
             instances_needed,
             self.max_live_instances - n_live_instances,
-            self.max_instances - self.n_instances,
+            self.max_instances - n_instances,
```

Comment:
Why did we make this change away from `self.n_instances` to the local variable? Should we remove the n_instances property in the base class now? Does it still have a sensible interpretation?

Reply:
The `self.n_instances` means the same thing. The issue is we're still computing the prometheus statistics for the full job queue as well as getting the prometheus statistics and then finally when we actually create instances. If you use the same dictionary for all calls to the `compute_n_instances` function, then your output will be wrong because the dictionary isn't mutated to account for new pending machines when you sequentially call the function for each region set.

I think we can get rid of the `full_job_queue` code and then get the prometheus statistics at the same time as creating the instances and that should simplify this.

Thoughts on getting rid of the full job queue statistics?